lib/services/elasticsearch.js

```diff
@@ -1360,218 +1241,213 @@ class ElasticSearch extends Service {
-      partialErrors = [],
-      metadata = {
-        _kuzzle_info: {
-          active: false,
-          deletedAt: Date.now(),
-          updater: this._getUserId(request)
-        }
-      };
-
-    esRequest.body = [];
+      esRequest = {
+        index: this._getESIndex(index, collection),
+        body: { ids: { values: [] } },
+        scroll: '30s',
+        refresh,
+        timeout
+      },
+      toGet = [],
+      toDelete = [],
+      partialErrors = [];
 
-    for (let i = 0; i < request.input.body.ids.length; i++) {
-      const id = request.input.body.ids[i];
+    for (let i = 0; i < ids.length; i++) {
+      const id = ids[i];
 
       if (typeof id === 'string') {
-        esRequest.body.push(
-          {update:
-            {_index: index,
-              _type: collection,
-              _id: id}
-          });
-        esRequest.body.push({doc: metadata});
-        ids.push([{_id: id}]);
+        toGet.push(id);
       } else {
-        partialErrors.push({id, reason: 'the document ID must be a string'});
+        partialErrors.push({ id, reason: 'the document ID must be a string' });
       }
     }
 
-    return this._mexecute(esRequest, ids, partialErrors)
-      .then(response => {
-        response.result = response.result.map(doc => doc._id);
-        return response;
-      });
-  }
+    return this.mGet(index, collection, toDelete)
+      .then(({ hits }) => {
+        let idx = 0;
 
-  /**
-   * This method can be called either:
-   *  - at the first creation of a collection : we need to apply the index default mapping
-   *  - when a collection mapping is updated : we need to protect the index common mapping
-   */
-  _mergeDefaultMapping (index, oldMapping, newMapping) {
-    const collectionMapping = {};
-    const commonMapping = _.cloneDeep(
-      this.kuzzle.indexCache.defaultMappings[index] ||
-      this.config.commonMapping
-    );
+        for (let i = 0; i < ids.length; i++) {
+          const
+            id = ids[i],
+            hit = hits[idx];
 
-    collectionMapping.dynamic = newMapping.dynamic ||
-    oldMapping.dynamic ||
-    this.config.dynamic;
-    collectionMapping._meta = newMapping._meta || oldMapping._meta || {};
-
-    // Preserve old version of kuzzle metadata mapping
-    if (oldMapping.properties && oldMapping.properties._kuzzle_info) {
-      Object.assign(
-        commonMapping._kuzzle_info.properties,
-        oldMapping.properties._kuzzle_info.properties
-      );
-    }
+          if (hit._id === id) {
+            toDelete.push(id);
+            idx++;
+          } else {
+            partialErrors.push({ id, reason: 'cannot find document' });
+          }
+        }
 
-    collectionMapping.properties = Object.assign({},
-      newMapping.properties,
-      commonMapping
-    );
+        esRequest.body.ids.values = toDelete;
 
-    return collectionMapping;
+        return this.client.deleteByQuery(esRequest);
+      })
+      .then(() => {
+        const result = {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "result". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1488)